aptos-move/e2e-move-tests/src/tests/fee_payer.rs

```diff
@@ -74,10 +77,129 @@ fn test_account_not_exist_with_fee_payer_create_account() {
     assert!(bob_start > bob_after);
 }
 
+#[test]
+fn test_account_not_exist_and_out_of_gas_with_fee_payer_create_account() {
+    // This should result in an aborted, yet committed transaction and an account created.
+    let mut h = MoveHarness::new_with_features(vec![FeatureFlag::GAS_PAYER_ENABLED], vec![]);
+
+    let alice = Account::new();
+    let bob = h.new_account_at(AccountAddress::from_hex_literal("0xb0b").unwrap());
+
+    let alice_start =
+        h.read_resource::<CoinStoreResource>(alice.address(), CoinStoreResource::struct_tag());
+    assert!(alice_start.is_none());
+    let bob_start = h.read_aptos_balance(bob.address());
+
+    let payload = aptos_stdlib::aptos_coin_transfer(*alice.address(), 1);
+    let transaction = TransactionBuilder::new(alice.clone())
+        .fee_payer(bob.clone())
+        .payload(payload)
+        .sequence_number(0)
+        .max_gas_amount(99_999) // This is not enough to execute this transaction
```

Comment:
This fails in validation (prologue), not inside user txn!

Reply:
This is failing during user txn. A sponsored transaction sending account does not need coin store. This attempts to access a coin store only during user txn. How to be sure? We were charged gas. That's only possible for transactions that succeed prologue. I do not think adding a more complex test improves testing. Maybe we have something I can copy over, but this test also demonstrates how powerful this concept is. It can reduce account costs for certain applications by 50%.

Reply:
That's the point. What's your point? The assertion is that it is discarded, i.e., dropped during prologue.

Reply:
For testing usually we want to cover success, out of gas, aborts. This test checks out of gas but during validation. There is no test that checks out of gas during user txn execution (see slack discussion).

It's not a problem of this particular PR, I think we don't have a good coverage around these corner cases for other features, and it seems an important case?

Reply:
yup... added!